portal/src/graphql/portal/CreateOAuthClientScreen.tsx

```diff
@@ -91,6 +91,9 @@ function constructConfig(
         draft.client_name = draft.name;
         draft.redirect_uris = ["http://localhost/after-authentication"];
         draft.post_logout_redirect_uris = undefined;
+      } else if (draft.x_application_type === "m2m") {
+        draft.redirect_uris = undefined;
+        draft.post_logout_redirect_uris = undefined;
```

Comment:
The created client application now looks like

```yaml
  - client_id: 84c509df70467cff
    grant_types:
    - authorization_code
    - refresh_token
    issue_jwt_access_token: true
    name: m2m
    response_types:
    - code
    - none
    x_application_type: m2m
```

It should not have `grant_types` and `response_types`.

---

When I search the keyword `grant_types`, I discovered `grant_types_supported`. We should add `client_credentials` to `grant_types_supported`.